test/integration/driver.go

```diff
@@ -81,342 +58,205 @@ var (
+	flag.StringVar(&params.infra.CaImage, "ca", "gcr.io/istio-testing/istio-ca:"+caTag,
 		"CA Docker image")
-	flag.StringVar(&params.mixerImage, "mixer", "gcr.io/istio-testing/mixer:"+mixerTag,
+	flag.StringVar(&params.infra.MixerImage, "mixer", "gcr.io/istio-testing/mixer:"+mixerTag,
 		"Mixer Docker image")
-	flag.StringVar(&params.namespace, "n", "",
+	flag.StringVar(&params.infra.Namespace, "n", "",
 		"Namespace to use for testing (empty to create/delete temporary one)")
 	flag.StringVar(&params.kubeconfig, "kubeconfig", "platform/kube/config",
 		"kube config file (missing or empty file makes the test use in-cluster kube config instead)")
 	flag.IntVar(&params.count, "count", 1, "Number of times to run the tests after deploying")
 	flag.StringVar(&authmode, "auth", "both", "Enable / disable auth, or test both.")
 	flag.BoolVar(&params.debug, "debug", false, "Extra logging in the containers")
-	flag.BoolVar(&params.parallel, "parallel", true, "Run requests in parallel")
 	flag.BoolVar(&params.logs, "logs", true, "Validate pod logs (expensive in long-running tests)")
 }
 
-func main() {
-	flag.Parse()
-	switch authmode {
-	case "enable":
-		params.auth = true
-		runTests()
-	case "disable":
-		params.auth = false
-		runTests()
-	case "both":
-		params.auth = false
-		runTests()
-		params.auth = true
-		runTests()
-	default:
-		glog.Infof("Invald auth flag: %s. Please choose from: enable/disable/both.", params.auth)
-	}
-}
-
-func runTests() {
-	glog.Infof("\n--------------- Run tests with auth: %t ---------------", params.auth)
-
-	setup()
-
-	for i := 0; i < params.count; i++ {
-		glog.Infof("Test run: %d", i)
-		check((&reachability{}).run())
-		check(testRouting())
-	}
-
+type test interface {
+	String() string
+	setup() error
+	run() error
 	teardown()
-	glog.Infof("\n--------------- All tests with auth: %t passed %d time(s)! ---------------\n", params.auth, params.count)
 }
 
-func setup() {
-	if params.tag == "" {
+func main() {
+	flag.Parse()
+	if params.infra.Tag == "" {
 		glog.Fatal("No docker tag specified")
 	}
-	if params.mixerImage == "" {
-		glog.Fatal("No mixer image specified")
-	}
-	glog.Infof("params %#v", params)
 
 	if params.debug {
-		params.verbosity = 3
+		params.infra.Verbosity = 3
 	} else {
-		params.verbosity = 2
+		params.infra.Verbosity = 2
 	}
 
 	check(setupClient())
 
-	var err error
-	if params.namespace == "" {
-		if params.namespace, err = generateNamespace(client); err != nil {
-			check(err)
-		}
-	} else {
-		_, err = client.Core().Namespaces().Get(params.namespace, meta_v1.GetOptions{})
-		check(err)
-	}
-
-	pods = make(map[string]string)
-
-	if params.auth {
-		check(deploy("ca", "ca", ca, "", "", "", "", "unversioned", false))
-		_, err = shell(fmt.Sprintf("kubectl -n %s apply -f test/integration/config-auth.yaml", params.namespace))
-	} else {
-		_, err = shell(fmt.Sprintf("kubectl -n %s apply -f test/integration/config.yaml", params.namespace))
+	params.infra.Mixer = true
```

Comment:
yeah, let's do that in another PR. we will need a way to select which tests via flags, not just for auth.